bhttp/src/http_compat/encode.rs

```diff
@@ -0,0 +1,441 @@
+            Ok::<_, std::convert::Infallible>(trailer_frame),
+        ]));
+
+        let response = Response::builder()
+            .status(200)
+            .header("Content-Type", "text/plain")
+            .body(body)
+            .unwrap();
+
+        let stream = BhttpEncoder::from_response(response);
+        let future = stream.fold(Vec::new(), |mut acc, item| async move {
+            if let Ok(data) = item {
+                acc.extend(data);
+            }
+            acc
+        });
+
+        let bhttp_data = block_on(future);
+
+        // Verify that we got some data
+        assert!(!bhttp_data.is_empty());
+    }
+
+    #[test]
+    fn test_encode_rfc9292_request_example_indeterminate_length() {
+        // Modifyed example from Section 5.1 of RFC 9292 - Indeterminate-Length Binary Encoding of Request
+        const REQUEST_EXAMPLE: &[u8] = &[
+            0x02, 0x03, 0x47, 0x45, 0x54, 0x05, 0x68, 0x74, 0x74, 0x70, 0x73, 0x09, 0x68, 0x65,
+            0x6c, 0x6c, 0x6f, 0x2e, 0x74, 0x78, 0x74, 0x01, 0x2f, 0x0a, 0x75, 0x73, 0x65, 0x72,
+            0x2d, 0x61, 0x67, 0x65, 0x6e, 0x74, 0x34, 0x63, 0x75, 0x72, 0x6c, 0x2f, 0x37, 0x2e,
+            0x31, 0x36, 0x2e, 0x33, 0x20, 0x6c, 0x69, 0x62, 0x63, 0x75, 0x72, 0x6c, 0x2f, 0x37,
+            0x2e, 0x31, 0x36, 0x2e, 0x33, 0x20, 0x4f, 0x70, 0x65, 0x6e, 0x53, 0x53, 0x4c, 0x2f,
+            0x30, 0x2e, 0x39, 0x2e, 0x37, 0x6c, 0x20, 0x7a, 0x6c, 0x69, 0x62, 0x2f, 0x31, 0x2e,
+            0x32, 0x2e, 0x33, 0x04, 0x68, 0x6f, 0x73, 0x74, 0x0f, 0x77, 0x77, 0x77, 0x2e, 0x65,
+            0x78, 0x61, 0x6d, 0x70, 0x6c, 0x65, 0x2e, 0x63, 0x6f, 0x6d, 0x0f, 0x61, 0x63, 0x63,
+            0x65, 0x70, 0x74, 0x2d, 0x6c, 0x61, 0x6e, 0x67, 0x75, 0x61, 0x67, 0x65, 0x06, 0x65,
+            0x6e, 0x2c, 0x20, 0x6d, 0x69, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+            0x00, 0x00, 0x00, 0x00,
+        ];
+
+        // Create the HTTP request as specified in the RFC
+        let request = Request::builder()
+            .method("GET")
+            .uri("https://hello.txt")
+            .header(
+                "user-agent",
+                "curl/7.16.3 libcurl/7.16.3 OpenSSL/0.9.7l zlib/1.2.3",
+            )
+            .header("host", "www.example.com")
+            .header("accept-language", "en, mi")
+            .body(Full::new(Bytes::new())) // No body for this request
+            .unwrap();
+
+        // Encode the request to BHTTP using our streaming encoder
+        let stream = BhttpEncoder::from_request(request);
+        let future = stream.fold(Vec::new(), |mut acc, item| async move {
+            if let Ok(data) = item {
+                acc.extend(data);
+            }
+            acc
+        });
+
+        let bhttp_data = block_on(future);
+
+        compare_encoded_data_with_the_expected_example(&bhttp_data, REQUEST_EXAMPLE);
+    }
+
+    fn compare_encoded_data_with_the_expected_example(generated: &[u8], expected: &[u8]) {
```

Comment:
This is a little simplistic.  In most cases, your code should generate the full message, including trailing zeros.  Or you should have a specific number of zeros to ignore.  There are some zero values that can't be omitted when encoding and this test will miss those.  For example, that includes the marker at the end of a field section.  If you had neglected to add that in your encoder, this test might miss that.

Reply:
I will remove the function and delete the 10 padding bytes from REQUEST_EXAMPLE, after which we can use `assert_eq!` directly to achieve an exact match.